File: yangxj96-starter/yangxj96-starter-security/src/main/java/io/github/yangxj96/starter/security/exception/SecurityException.java
/*****************************
 * Copyright (c) 2021 - 2023
 * author:yangxj96
 * email :yangxj96@gmail.com
 * date  :2023-01-07 00:11:06
 * Copyright (c) 2021 - 2023
 ****************************/

package io.github.yangxj96.starter.security.exception;

/**
 * 安全权限异常
 *
 * @author yangxj96
 * @version 1.0
 * @date 2023-01-07 00:14
 */
public class SecurityException extends RuntimeException {

    public SecurityException() {
        super();
    }

    public SecurityException(String message) {
        super(message);
    }

    public SecurityException(String message, Throwable cause) {
        super(message, cause);
    }

    public SecurityException(Throwable cause) {
        super(cause);
    }

    protected SecurityException(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
        super(message, cause, enableSuppression, writableStackTrace);
    }
}

File: yangxj96-starter/yangxj96-starter-security/src/main/java/com/yangxj96/saas/starter/security/exception/SecurityException.kt
package com.yangxj96.saas.starter.security.exception

/**
 * 安全权限异常
 */
open class SecurityException : RuntimeException {
    constructor() : super()
    constructor(message: String?) : super(message)
    constructor(message: String?, cause: Throwable?) : super(message, cause)
    constructor(cause: Throwable?) : super(cause)
    protected constructor(
        message: String?,
        cause: Throwable?,
        enableSuppression: Boolean,
        writableStackTrace: Boolean
    ) : super(message, cause, enableSuppression, writableStackTrace)
}
